test/IDE/complete_macros_expanded.swift

```diff
@@ -0,0 +1,79 @@
+
+// RUN: %target-swift-ide-test -batch-code-completion -source-filename %s -filecheck %raw-FileCheck -completion-output-dir %t -plugin-path %t/plugins -parse-as-library 
+
+@freestanding(declaration, names: named(A), named(B), named(foo), named(addOne))
+macro defineDeclsWithKnownNames() = #externalMacro(module: "MacroDefinition", type: "DefineDeclsWithKnownNamesMacro")
+
+@attached(peer, names: suffixed(_peer))
+macro PeerWithSuffix() = #externalMacro(module: "MacroDefinition", type: "PeerValueWithSuffixNameMacro")
+
+#defineDeclsWithKnownNames
+
+@PeerWithSuffix
+func globalFunc() {}
+
+func test() {
+   #^GLOBAL^#
+// GLOBAL-DAG: Decl[Struct]/CurrModule:            A[#A#]; name=A
+// GLOBAL-DAG: Decl[Struct]/CurrModule:            B[#B#]; name=B
+// GLOBAL-DAG: Decl[GlobalVar]/CurrModule:         foo[#Int#]; name=foo
+// GLOBAL-DAG: Decl[GlobalVar]/CurrModule:         addOne[#(Int) -> Int#]; name=addOne
+// GLOBAL-DAG: Decl[FreeFunction]/CurrModule:      globalFunc()[#Void#]; name=globalFunc()
+// GLOBAL-DAG: Decl[GlobalVar]/CurrModule:         globalFunc_peer[#Int#]; name=globalFunc_peer
+}
+
+struct MyStruct {
+  @PeerWithSuffix
+  func instanceMethod() {}
+
+  @PeerWithSuffix
+  static func staticMethod() {}
+
+  #defineDeclsWithKnownNames
+}
+
+func testMember(value: MyStruct) {
+  value.#^MEMBER_INSTANCE^#
+// MEMBER_INSTANCE-DAG: Keyword[self]/CurrNominal:          self[#MyStruct#]; name=self
+// MEMBER_INSTANCE-DAG: Decl[InstanceVar]/CurrNominal:      foo[#Int#]; name=foo
+// MEMBER_INSTANCE-DAG: Decl[InstanceVar]/CurrNominal:      addOne[#(Int) -> Int#]; name=addOne
+// MEMBER_INSTANCE-DAG: Decl[InstanceMethod]/CurrNominal:   instanceMethod()[#Void#]; name=instanceMethod()
+// MEMBER_INSTANCE-DAG: Decl[InstanceVar]/CurrNominal:      instanceMethod_peer[#Int#]; name=instanceMethod_peer
+// MEMBER_INSTANCE-DAG: Decl[InstanceVar]/CurrNominal:      staticMethod_peer[#Int#]; name=staticMethod_peer
+// NOTE: 'staticMethod_peer' is a instance var because the macro emits the decl without 'static'
+
+  MyStruct.#^MEMBER_STATIC^#
+// MEMBER_STATIC-NOT: _peer
+// MEMBER_STATIC-DAG: Keyword[self]/CurrNominal:          self[#MyStruct.Type#]; name=self
+// MEMBER_STATIC-DAG: Keyword/CurrNominal:                Type[#MyStruct.Type#]; name=Type
+// MEMBER_STATIC-DAG: Decl[Struct]/CurrNominal:           A[#MyStruct.A#]; name=A
+// MEMBER_STATIC-DAG: Decl[Struct]/CurrNominal:           B[#MyStruct.B#]; name=B
+// MEMBER_STATIC-DAG: Decl[InstanceMethod]/CurrNominal:   instanceMethod({#(self): MyStruct#})[#() -> Void#]; name=instanceMethod(:)
+// MEMBER_STATIC-DAG: Decl[StaticMethod]/CurrNominal:     staticMethod()[#Void#]; name=staticMethod()
+// MEMBER_STATIC-NOT: _peer
+}
+
+func testLocal() {
+  #defineDeclsWithKnownNames
+
+  @PeerWithSuffix func localFunc() {}
+
+  do {
+    #^LOCAL?skip=FIXME^#
```

Comment:
Change this to an `xfail` so we don’t forget to change it if it’s fixed?
```suggestion
    #^LOCAL?xfail=FIXME^#
```

Reply:
Ah, OK. Makes sense.